exonum/src/runtime/oneshot.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A channel for sending a deployment status between threads.
+
+use std::sync::mpsc;
+
+use super::{ErrorKind, ExecutionError};
+
+/// Create a new one-shot channel for sending deployment status,
+/// returning the sender/receiver halves.
+pub fn channel() -> (Sender, Receiver) {
+    let (tx, rx) = mpsc::channel();
+    (Sender(tx), Receiver(rx))
+}
+
+/// The receiving half of [`channel`] type. This half can only be owned by one thread.
+///
+/// [`channel`]: fn.channel.html
+#[derive(Debug)]
+pub struct Receiver(mpsc::Receiver<Result<(), ExecutionError>>);
+
+impl Receiver {
+    /// Attempts to wait for a value on this receiver, returning an error if the
+    /// corresponding channel has hung up.
+    pub(crate) fn wait(self) -> Result<(), ExecutionError> {
+        self.0.recv().map_err(|e| {
```

Comment:
Nit: I don't think it makes sense to output an error; [`RecvError`](https://doc.rust-lang.org/std/sync/mpsc/struct.RecvError.html) is an empty struct, so it does not provide any details as to the error cause.